Feature: payment

NicolasRichel opened the pull request:
As discussed yesterday, I would like to merge a first (unfinished) draft with **payment** features that are already implemented.
This includes:
 - Organizations Management ~(still some bugs)~
 - Subscriptions view (with the list of billings and invoices)
 - "Upgrade Storage" buttons on Space board and Project board

The Payment view (with payment form) is not implemented yet.

Amoki committed:
WIP: first Paddle implementation

Amoki committed:
wip: test some stuff

LrxGaelle committed:
add PlatformSubscription page & header link

LrxGaelle committed:
add Invoices specific component

LrxGaelle committed:
add BillingDetails specific component

LrxGaelle committed:
add OurPlans specific component

LrxGaelle committed:
add BIMDataUtilities & BIMDAtaFonts to global Platform

LrxGaelle committed:
add EmptyBilling icon

LrxGaelle committed:
add PlatformSubscription to router

LrxGaelle committed:
add upgrade storage btn & translation

LrxGaelle committed:
add translations for Dashboard Platform Pro btn

LrxGaelle committed:
add translations on the platform's payment button on the dashboard

LrxGaelle committed:
rebase from develop + retrieve organizations from private api

LrxGaelle committed:
add organizations list to PlatformSubscription

NicolasRichel committed:
feat(state): create organization service/state

NicolasRichel committed:
feat(state): little reword + load user organizations in root resolver

LrxGaelle committed:
Feature/platform subscription card (#72)

* wip: test some stuff

* WIP: first Paddle implementation

* add example to set prices

* add PlatformSubscription page & header link

* add PlatformSubscription to router

* add upgrade storage btn & translation

* create PaddlePayment view

* create Dasboard Subcription Platform card

* create payment state & paymentService

* delete unused function into Dashboard view

* space removal

* delete duplicate functions on SpaceBoard view

* add create, update, delete methods for organizations

* delete unused style into DashboardButtonTile.scss

Co-authored-by: amoki <hugo@bimdata.io>

NicolasRichel committed:
test(state): write some tests for organization state

NicolasRichel committed:
fix: reformat SpaceBoard component

NicolasRichel committed:
feat(organizations): create organization management panel on user spaces view

NicolasRichel committed:
feat(orga): create orga form + orga spaces manager (wip)

LrxGaelle committed:
Feature/subscription view (#76)

* WIP: first Paddle implementation

* add example to set prices

* wip: test some stuff

* WIP: first Paddle implementation

* add example to set prices

* add PlatformSubscription page & header link

* add Invoices specific component

* add BillingDetails specific component

* add OurPlans specific component

* add BIMDataUtilities & BIMDAtaFonts to global Platform

* add EmptyBilling icon

* add PlatformSubscription to router

* add upgrade storage btn & translation

* add translations for Dashboard Platform Pro btn

* add translations on the platform's payment button on the dashboard

* rebase from develop + retrieve organizations from private api

* add organizations list to PlatformSubscription

* WIP: first Paddle implementation

* add example to set prices

* wip: test some stuff

* WIP: first Paddle implementation

* add example to set prices

* add PlatformSubscription page & header link

* add Invoices specific component

* add BillingDetails specific component

* add OurPlans specific component

* add BIMDataUtilities & BIMDAtaFonts to global Platform

* add EmptyBilling icon

* add PlatformSubscription to router

* add upgrade storage btn & translation

* add translations for Dashboard Platform Pro btn

* add translations on the platform's payment button on the dashboard

* rebase from develop + retrieve organizations from private api

* add organizations list to PlatformSubscription

* feat(state): create organization service/state

* feat(state): little reword + load user organizations in root resolver

* Feature/platform subscription card (#72)

* wip: test some stuff

* WIP: first Paddle implementation

* add example to set prices

* add PlatformSubscription page & header link

* add PlatformSubscription to router

* add upgrade storage btn & translation

* create PaddlePayment view

* create Dasboard Subcription Platform card

* create payment state & paymentService

* delete unused function into Dashboard view

* space removal

* delete duplicate functions on SpaceBoard view

* add create, update, delete methods for organizations

* delete unused style into DashboardButtonTile.scss

Co-authored-by: amoki <hugo@bimdata.io>

* WIP: first Paddle implementation

* add example to set prices

* wip: test some stuff

* WIP: first Paddle implementation

* add example to set prices

* add PlatformSubscription page & header link

* add Invoices specific component

* add BillingDetails specific component

* add OurPlans specific component

* add BIMDataUtilities & BIMDAtaFonts to global Platform

* add EmptyBilling icon

* add PlatformSubscription to router

* add upgrade storage btn & translation

* add translations for Dashboard Platform Pro btn

* add translations on the platform's payment button on the dashboard

* rebase from develop + retrieve organizations from private api

* add organizations list to PlatformSubscription

* feat(state): create organization service/state

* feat(state): little reword + load user organizations in root resolver

* Feature/platform subscription card (#72)

* wip: test some stuff

* WIP: first Paddle implementation

* add example to set prices

* add PlatformSubscription page & header link

* add PlatformSubscription to router

* add upgrade storage btn & translation

* create PaddlePayment view

* create Dasboard Subcription Platform card

* create payment state & paymentService

* delete unused function into Dashboard view

* space removal

* delete duplicate functions on SpaceBoard view

* add create, update, delete methods for organizations

* delete unused style into DashboardButtonTile.scss

Co-authored-by: amoki <hugo@bimdata.io>

* add selectedOrganization & organization length

* add style for organizations length

* add selectedOrganization default value & sort organizations list

* add selectedOrganization into PlatformSubscription view

* delete unused file

* add retrieveOrganizationPlatformSubscriptions service / state

* update bimdata/design-system

* add organization Billing details

* custom organization Billing details

* add billing details translations

* add OurPlans translations

* add Invoices title translation

* add formatDate

* format columns file

* add platformSubscriptionPayments

* add subscriptions to Invoices

* create invoices table

* move date.js utils

* change status header labels

* change invoices columns

* change formatDate import

* replace forEach by reduce for subscriptionsMap

* add translations

* add goToPaddlePayment methods

* custom deleted status in billing details view

* empty card billing custom

* delete unused status custom

* don't show billing details actions if billing deleted

* change status label for billing details table

* fix: put emptyBilling.svg into static assets folder

Co-authored-by: amoki <hugo@bimdata.io>
Co-authored-by: NicolasRichel <nicolas@bimdata.io>

NicolasRichel committed:
chore: some fixes after rebase

NicolasRichel committed:
feat(orga): create organization space import

NicolasRichel committed:
chore: update dependencies (using ncu)

NicolasRichel committed:
feat(orga): organization space import

NicolasRichel committed:
feat(orga): organization update form (wip)

NicolasRichel committed:
feat(orga): refactor component structure for OrganizationsManager

NicolasRichel committed:
feat(orga): handle create/update scenario in OrganizationForm

NicolasRichel committed:
feat(orga): fix orga creation/update/import

NicolasRichel committed:
feat(orga): some fixes in orga form and spaces import

NicolasRichel committed:
feat(orga): create a mock impl for OrganizationDeleteGuard

LrxGaelle committed:
format SpaceBoard file

LrxGaelle committed:
update package-lock

LrxGaelle committed:
Feature/progress bar (#78)

* create ProgressBar component

* dissociate the progress bar if we are in a paid space or not

* add formatBytes utils

* add space size service / state

* add retrievePlaformSubscriptions state

* custom progress bar component with
- add computed for indicatorColorState
- rename progressPercent props
- custom text size

* use progressBar into SpaceBoard

* add condition for progressPercent in progressBar component

* add translations for progressBar component

* add conditions for display progress bar

* refacto progressBar in SpaceBoard

* create StorageInformations specific component

* use StorageInformations component

* fix: rearrange file utils

NicolasRichel committed:
fix(styles): remove those painful scrollbars from project view

NicolasRichel committed:
fix(orga): fix some styles + orga spaces import

LrxGaelle committed:
WIP (payment) : dev on payment view - synchro commit

Amoki commented on src/components/specific/organizations/organizations-manager/organization-spaces-list/organization-space-card/OrganizationSpaceCard.scss:
```diff
@@ -0,0 +1,40 @@
+      &--date {
+        color: var(--color-tertiary-dark);
+
+        // &::before {
```
dead code ?

Amoki commented on src/utils/files.js:
```diff
@@ -16,11 +19,24 @@ function formatBytes(bytes) {
   return `${bytes} B`;
 }
 
+function formatBytesV2(bytes, decimals = 2) {
```
Do we want to keep `formatBytes()`? This one seems better in all cases.
And it would avoid this weird naming

Amoki commented on src/views/space-board/SpaceBoard.vue:
```diff
@@ -15,7 +15,9 @@
-        <!-- <BIMDataButton
+        <div class="flex items-center">
+          <StorageInformations :space="space" />
+          <!-- <BIMDataButton
```
dead code ?

NicolasRichel committed:
fix(orga): in spaces import, replace submit button at the bottom of the list

NicolasRichel committed:
feat(payment): fix paddle setup + create subscription payment UI skeleton

LrxGaelle committed:
change router name for StorageInformation component

LrxGaelle committed:
add DropdownList for selected orga & space in SpacePayment view

NicolasRichel committed:
feat(payment): dev on payment view

NicolasRichel committed:
feat(payment): refactor css on payment view

NicolasRichel committed:
refavtor: rename PlatformSubscription view to UserSubscriptions

NicolasRichel committed:
refactor: create PlatformSubscription sub view and use it in Payment view

NicolasRichel committed:
refactor: setup DatapackSubscription view and add it to Payment view

NicolasRichel committed:
feat(payment): rewrite platform-sub-info component

NicolasRichel committed:
refactor: rename dashboard-platform-sub to dashboard-button-tile and fix i18n

LrxGaelle committed:
create organization & space selectors

LrxGaelle committed:
rename '__content' class into '__selectors'

LrxGaelle committed:
delete unused params into StorageInformations view

NicolasRichel committed:
refactor: start scenario detection logic for payment view

NicolasRichel committed:
chore: TODO comment for Jaelle

NicolasRichel committed:
feat(payment): add space selection logic in Payment view (script only)

NicolasRichel committed:
refactor: rename/reformat some components + make sure there are no routing to space-payment anymore

NicolasRichel committed:
feat(payment): dev on payment view initialization

NicolasRichel committed:
chore: update api-client version to 6.7.2

NicolasRichel committed:
feat(payment): rearrange payment view init

LrxGaelle committed:
create orga and space selectors

NicolasRichel committed:
refactor(payment): rename payment state and methods + fix user-subscriptions + create datapack components

LrxGaelle committed:
replace flex by grid for DataPack & Platform form

LrxGaelle committed:
add box-shadow to BIMDataPlatform heading

LrxGaelle committed:
add translations for OrgaSpaceCards & OrgaSpaceSelectors components

LrxGaelle committed:
create OrgaSpaceCards component

LrxGaelle committed:
adjust create OrgaSpaceSelectors component

- add conditionnal bloc if no space paid
- add text if no space paid
- add margin

LrxGaelle committed:
custom Payment vue

- add space filter for DataPack view (only space paid)
- add orga filter (all organizations except personnal)
- add OrgaSpaceCards component

NicolasRichel committed:
feat(payment): implement datapack-sub-form + some refacto

NicolasRichel committed:
feat(payment): integrate creator/selector into payment views

NicolasRichel committed:
feat(payment): finish routing integration + refactor space-creator

NicolasRichel committed:
feat(payment): finalize subscription-platform view

NicolasRichel committed:
refactor(payment): dev on datapack sub view + refactor payment state and change it to subscriptions state

NicolasRichel committed:
refactor(payment): create paddle composable to centralize Paddle calls

NicolasRichel committed:
fix(orga): update orga spaces list when creating/updating/deleting a space

NicolasRichel committed:
feat(payment): fix and finish datapack sub form

NicolasRichel committed:
feat(payment): finalize platform subscription form

NicolasRichel committed:
feat(payment): disable ifc/file upload when plan storage is fully used

NicolasRichel committed:
feat(components): improve progress bar by adding waringLimit and highLimit props

NicolasRichel committed:
feat(payment): create subscribe view to choose sub plan

NicolasRichel committed:
feat(payment): create subscription-free view

NicolasRichel committed:
refactor(payment): rename sub-platform to sub-pro

NicolasRichel committed:
refactor(payment): rename platform-sub and datapack components

NicolasRichel committed:
refactor: move 'payment' components to 'subscriptions'

NicolasRichel committed:
feat(payment): add ability to disable subscription features

NicolasRichel committed:
feat(payment): refactor + dev on user-subscriptions view

NicolasRichel committed:
feat(payment): change dashboard subscriptions tile

NicolasRichel committed:
feat(payment): put plans storage values into env + get price data from Paddle instead of hard coding them

NicolasRichel committed:
feat(dashboard): add subscribe card to dashboard

NicolasRichel committed:
chore: update api-client version

NicolasRichel committed:
feat(subscription): implement susbcribe guard

NicolasRichel committed:
feat(payment): use new space creation mechanism

Amoki committed:
add automatic deployment for this branch

NicolasRichel committed:
feat(payment): hide orga manager button when subscription is disabled

NicolasRichel committed:
Merge branch 'feature/payment' of github.com:bimdata/platform-next into feature/payment

NicolasRichel committed:
fix(config): put subscription config (from env) into a config module

NicolasRichel committed:
fix(config): fix subscription config for routing

NicolasRichel committed:
fix(config): try to use an iife to properly take subscription env value into account (when deployed)

NicolasRichel committed:
fix(config): try to define a function to use subscription env value

NicolasRichel committed:
fix(config): add some debug logs

NicolasRichel committed:
chore(deployment): add subscription env vars to Dockerfile and env.sh

NicolasRichel committed:
chore: remove debug logs

NicolasRichel committed:
refactor(state): rename some methods + little rearrangements

NicolasRichel committed:
refactor(state): [wip] remove soft update logic and reload data from API instead

NicolasRichel committed:
refactor(state): remove soft update logic for models & files states (soft delete still remain)

NicolasRichel committed:
refactor(state): remove soft update from space state

NicolasRichel committed:
Merge branch 'develop' into feature/payment

NicolasRichel committed:
fix(views): add go-back-button to user-subs view

NicolasRichel committed:
feat(subscription): add 'free badge' to free spaces cards

NicolasRichel committed:
feat(subscription): enable free space creation when subscription is disabled

NicolasRichel committed:
fix(subscription): adapt space-creator for free subscription scenario

NicolasRichel committed:
fix(subscription): various fixes on styles, routing and components behavior

NicolasRichel committed:
fix(datapack): go back to previous view when datapack successfully created/updated

NicolasRichel committed:
fix(routing): check user is authenticated before loading data

NicolasRichel committed:
fix(routing): redirect free plan to pro plan if user have already a free space

NicolasRichel committed:
fix(styles): add hover effect on organization cards

NicolasRichel committed:
Merge branch 'develop' into feature/payment

NicolasRichel committed:
feat(routing): update go-back-button behavior + refactor resolvers

NicolasRichel committed:
Merge branch 'develop' into feature/payment

NicolasRichel committed:
fix(date): use new date formatting for payment features

NicolasRichel committed:
chore: show imports file extension

NicolasRichel committed:
refactor(state): extract state mappers into a separate file

NicolasRichel committed:
fix: i18n for paddle checkout + fix some styles

NicolasRichel committed:
Merge branch 'develop' into feature/payment

NicolasRichel committed:
fix(styles): update color variables names for payment

NicolasRichel committed:
feat(user-subs): auto-select orga when coming from a space/project

NicolasRichel committed:
feat(user-subs): access space-board from billing-table

NicolasRichel committed:
fix(billings-table): show subscription total price (including datapacks)

NicolasRichel committed:
feat(billings-table): add 'Manage datapacks' button to billing action menu

NicolasRichel committed:
fix(sub-pro): create orga on space name submition

NicolasRichel committed:
reafctor: create space-selected component to display current space on subs views

NicolasRichel committed:
refactor(datapack): start refactoring of subscription-datapack view

NicolasRichel committed:
refactor(datapack): refcator datapack form component

NicolasRichel committed:
Merge branch 'develop' into feature/payment

NicolasRichel committed:
feat(datapack): add cancel button action + change view title

NicolasRichel committed:
refactor: turn svg asset into vue compoenents and group them in 'components/images' folder

NicolasRichel committed:
refactor: use BIMDataFileIcon component from design-system

NicolasRichel committed:
chore: update dependencies (using ncu)

NicolasRichel committed:
fix: bullet marks in pro-plan-info

NicolasRichel committed:
fix(datapack): ensure the currently active datapack is selected when updating

NicolasRichel committed:
fix(subscription): avoid space duplication when upgrading a free space to pro plan

NicolasRichel committed:
fix(styles): some various css fixes

NicolasRichel committed:
feat(components): use BIMDataTextBox component from design-system

NicolasRichel committed:
feat(payment): update/fix/complete payment wording

NicolasRichel committed:
fix(datapack): wait for updated space size on datapack submit before routing to space-board

NicolasRichel committed:
Merge branch 'develop' into feature/payment

NicolasRichel commented:
Payment Feature finalized, ready to be merged. :)

Bimdata-io commented:
:tada: This PR is included in version 1.0.0 :tada:

The release is available on [GitHub release](https://github.com/bimdata/platform/releases/tag/v1.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
